Allowing Opdam 2017 to run on vanilla toolbox

akaviaLab opened the pull request:
I've been trying to run the analysis described in the Opdam 2017 paper on additional models. I've been comparing the results from the published code to the standard toolbox.
GIMME and FASTCORE give different results since the Opdam 2017 uses older code, which has had bugs which were fixed.
mCADRE behaves too different to match up, probably due to different precursor metabolite definition
MBA works fine, but looking at the Opdam code I speeded up consistency Checking a bit. Since MBA removes random reactions, the entire model can become infeasible, and in that case, no need to run through the complete fastcc loops (of whatever variant).
INIT and iMAT work fine and give identical results except for the constrained model, which uses much smaller epsilon than the default (not sure why it needs that to work). I've added the option to give epsilon as a parameter to INIT and iMAT.

**I hereby confirm that I have:**

- [X] Tested my code on my own machine
- [X] Followed the guidelines in the [Contributing Guide](https://opencobra.github.io/cobratoolbox/docs/contributing.html)
- [X] Selected `develop` as a target branch (top left drop-down menu)

laurentheirendt commented on src/dataIntegration/transcriptomics/INIT/INIT.m:
```diff
@@ -1,4 +1,4 @@
-function tissueModel = INIT(model, weights, tol, runtime, logfile)
+function tissueModel = INIT(model, weights, tol, runtime, logfile, epsilon)
```
May you also update the documentation header with this new optional input?

laurentheirendt commented on src/dataIntegration/transcriptomics/iMAT/iMAT.m:
```diff
@@ -1,4 +1,4 @@
-function tissueModel = iMAT(model, expressionRxns, threshold_lb, threshold_ub, tol, core, logfile, runtime)
+function tissueModel = iMAT(model, expressionRxns, threshold_lb, threshold_ub, tol, core, logfile, runtime, epsilon)
```
Same comment about the documentation header here.

arichelle commented:
@akaviaLab it's unclear to me what the problems are. When to say that you want to run the analysis of Opdam 2017, do you talk about the code available in the github repo LewisLabUCSD/Context_Specific_Models_from_GeMS ? 
Please do not hesitate to contact me to assess these problems

akaviaLab commented:
Hi Anne,

I tried using the Github code
(LewisLabUCSD/Context_Specific_Models_from_GeMS) to rerun your analysis as
published, with the goal of running it on other models. I've come across
three types of problems
1) The code on github does not allow to recreate your results. Examples
include parsed_GPR_*.mat not available, removeNonUsedGenes having bugs, and
solveCobraMILP_loc not able to deal with newer versions of Gurobi.
2) Modifications needed to allow this to run on other metaoblic models
(such as Recon 2.0, Recon 2.2) - this includes constrainModel added as a
new function, and changing removeIsoforms. I also changed the behavior for
calculating gene expression a bit.
3) Differences between the github code and the local toolbox - I've been
working on comparing the results on Recon1. The summary of this comparison
is present in the beginning of the pull request. I've also added some
changes to INIT and iMAT to make sure that the toolbox and the github code
give identical results.

You can look at my github fork of the code, and the changes I've made. I'd
be happy to discuss it if there are specific questions.

Yours,

Uri David

On Tue, May 15, 2018 at 12:40 PM, Laurent Heirendt <notifications@github.com
> wrote:

> *@laurentheirendt* approved this pull request.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/opencobra/cobratoolbox/pull/1199#pullrequestreview-120318498>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AKGMZjkro3I-DrrRY1zqzOY6OIZbN6oKks5tywTugaJpZM4T5FZl>
> .
>



-- 
Uri David Akavia, Ph.D.
Assistant Professor, Biochemistry Department
Associate Member, Rosalind & Morris Goodman Cancer Research Centre
Membre Associé, Centre de recherche sur le cancer Rosalind & Morris Goodman
uri.david.akavia@mcgill.ca
McIntyre Medical Sciences Building
3655 Promenade Sir William Osler, Room 815B
Montreal, Quebec H3G 1Y6
Office tel: 514-398-8526
*Facebook <https://www.facebook.com/GCRC.McGill/>* | *Twitter*
<https://twitter.com/McGillGCRC> | *LinkedIn*
<https://www.linkedin.com/company/goodman-cancer-research-centre-mcgill-university/>
 | *Instagram <http://www.instagram.com/goodmancancerresearchcentre>** | **Gala
<http://mcgillgcrcgala.com/>** | **Défi Canderel
<https://www.deficanderel.com/>*
[image: cid:image003.jpg@01D3E5DF.2BE3BF50]

akaviaLab commented:
Anne, can you please confirm this can be merged?

arichelle commented:
Yes, the introduction of epsilon as an additional parameter for INIT and iMAT can be merged

laurentheirendt commented:
Thanks @akaviaLab and @arichelle 👍

akaviaLab commented:
Oops, I didn't update the documentation in createTissueSpecificModel. How shall we go about this? New PR? Push to the existing branch? Somebody else add it?
